src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -492,6 +477,16 @@ public boolean isFreePos(final BlockPos block)
         return freePositions.contains(block);
     }
 
+    @Override
+    public void invalidateRecipes(@NotNull final RecipeManager recipeManager)
+    {
+        // TODO: this should probably also invalidate anything using tags too (which is most things)
+        //       although technically there's a different event for tag reloads
+
+        compostRecipes.clear();
+        discoverCompostRecipes(recipeManager);
+    }
+
```

Comment:
This is called automatically on datapack reload to regenerate the cache, as I already described in the Discord.